src/plugins/saved_objects_tagging_oss/public/decorator/inject_tag_references.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObjectConfig } from '../../../saved_objects/public';
+import { InternalTagDecoratedSavedObject } from './types';
+
+/**
+ * Inject the tags back into the object's references
+ *
+ * (`injectReferences`) is used when fetching the object from the backend
+ */
+export const injectTagReferences: Required<SavedObjectConfig>['injectReferences'] = (
+  object,
+  references = []
+) => {
+  ((object as unknown) as InternalTagDecoratedSavedObject).__tags = references
+    .filter(({ type }) => type === 'tag')
```

Comment:
a side question: do we know all possible `type` values to declare it as an `union`?

Reply:
Unfortunately we don't, as types are dynamically registered from the server-side `registerType` API.